### DISCUSSION TITLE
accept is not a function

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1421] [Q&A] [answered]
when i want to use validate ,there reports an error
![image](https://github.com/eclipse-langium/langium/assets/26995040/1716c03b-f38e-4025-8743-81cb8f16be66)

![image](https://github.com/eclipse-langium/langium/assets/26995040/31a0c43e-1700-4db2-af06-73e10f8cbb6e)

**Comment by msujew** [comment]
@zrwsmd Can you provide a reproducible example? It works in all of our projects.

**Reply by zrwsmd** [reply]
open with vscode,and press f5 to launch extension,then error occured

**Reply by zrwsmd** [reply]
[glsp-vscode-integration-master-without-glsp-0319-half.zip](https://github.com/eclipse-langium/langium/files/14647127/glsp-vscode-integration-master-without-glsp-0319-half.zip)

**Reply by msujew** [reply, marked as answer]
The module hasn't been properly defined:

```diff
export function createStatemachineModule(
    context: StatemachineModuleContext
): Module<StatemachineServices, PartialLangiumServices & StatemachineAddedServices> {
    return {
        shared: () => context.shared,
        validation: {
-            StatemachineValidator: new StatemachineValidator()
+            StatemachineValidator: () => new StatemachineValidator()
        }
    };
}
```

**Reply by zrwsmd** [reply]
thanks a lot ,it truly works,but how can i pass this validation message to outer ,i mean  when vscode extension start,and open below file ,my custom validation message  'program name should not start with a digital' hint can show in the file
![image](https://github.com/eclipse-langium/langium/assets/26995040/e8203295-b55a-4fa3-a464-fb9cd1324077)

**Reply by zrwsmd** [reply]
i try to get from document.diagnostics ,but it is undefined
